### PR TITLE
feat: signup in unity

### PR DESCRIPTION
**Pull request by AjimenezDCL**
#408 

Implement the signup process in Unity

### CODE REVIEW

**Comment by AjimenezDCL** [comment]
This branch can be previewed at  https://play.decentraland.zone/branch/feat/signup_screen_unity/index.html?renderer=urn:decentraland:off-chain:renderer-artifacts:feat/signup_in_unity&ENV=org

Remember that must be tested alongside this PR: https://github.com/decentraland/explorer/pull/2401

**Review by kuruk-mm — COMMENTED** [review]
In the SignupHUD, when we press "TAB", it doesn't changes between email and name.

And when we go to the end of the terms, the last sentence it's a bit cut off.
image

After that, everything looks good.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public enum DCLAction_Trigger
     BuildEditShowAllEntities = 445,
     BuildEditModeResetCamera = 446,
     BuildEditModeZoomIn = 447,
-    BuildEditModeZoomOut = 448
+    BuildEditModeZoomOut = 448,
```
Comment (pravusjif) [inline unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/InputController/InputController.cs:69]: This is probably not needed right?

**Review by pravusjif — APPROVED** [review]
LGTM, the only concern I have is that, sorry I know I said before that the T&C mac sensibility was fine, but now re-testing it I think it's super fast, can we halve its speed for mac users?

**Review by kuruk-mm — APPROVED** [review]
I think the Navigation with "TAB" will be in another issue.

Another little thing, it's when we put the cursor hover the text boxes it will be nice to have the `I` cursor and and the same to the button and a hand cursor.

Nevertheless, looks great.

File: unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SignupHUD/SignupHUDController.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+        internal virtual ISignupHUDView CreateView() => SignupHUDView.CreateView();
+
+        public void Initialize(IHUD avatarEditorHUD)
+        {
+            view = CreateView();
+            if (view == null)
+                return;
+
+            this.avatarEditorHUD = avatarEditorHUD;
```
[the remark this answers is not in the document]
Comment (AjimenezDCL) [reply unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/HUD/SignupHUD/SignupHUDController.cs:23]: This is in fact an `avatarEditorHUD` hahahaha. We need a reference to it to open it back with the `Edit Avatar` button. Luckily I only needed to reference the common `IHud` so we avoid referencing directly the `AvatarEditorHUD` assembly.

**Comment by AjimenezDCL** [comment]
> I think the Navigation with "TAB" will be in another issue.
> 
> Another little thing, it's when we put the cursor hover the text boxes it will be nice to have the `I` cursor and and the same to the button and a hand cursor.
> 
> Nevertheless, looks great.

@kuruk-mm The interactive mouse is not a built-in feature from Unity, we can implement it but it should be a project-wide solution and in another PR.

**Comment by sandrade-dcl** [comment]
Be carefoul with the screen resolutions. With a resolution of 1280x1024 I'm seeing the "Tell us more about you" and "Terms of Service" panels cutted in the right side:

image
image

**Review by sandrade-dcl — APPROVED** [review]

**Comment by pravusjif** [comment]
Can @RominaMarchetti take a look at these changes before we merge them and get them to production?

**Comment by AjimenezDCL** [comment]
If I'm not mistaken Romi already gave her approval, in any case we need first the kernel PR

**Comment by RominaMarchetti** [comment]
@AjimenezDCL 
- Be careful with the feedback on the "Tell Us More". The active text box should have a grey stroke. It turns to red when the name reached the 15 characters (the counter should change to red as well), or if the email is not valid (the feedback message under the input is missing as well).
- The scroll in "Terms and conditions" with the trackpad feels extremely fast.
- The position of the cards, titles, buttons, sizes, should be the same between the two screens.
- Check text fonts, styles and size.

Figma project: https://www.figma.com/file/feevEf3dpoEMDLLcm5deSs/SignInV1?node-id=669%3A0

**Pull request merged**